src/extension.ts

```diff
@@ -24,6 +25,47 @@ export function activate(context: vscode.ExtensionContext) {
 		(selectionOverride?: vscode.Selection) => continueEditing(outline, false, selectionOverride)
 	));
 
+	context.subscriptions.push(vscode.commands.registerCommand(COMMAND_MARKDOWN_COPILOT_EDITING_NAME_AND_SAVE, async () => {
+		const textEditor = vscode.window.activeTextEditor;
+		if (textEditor === undefined) { return; }
+
+		function getDate(): string {
+			const date = new Date();
+			const year = date.getFullYear();
+			const month = String(date.getMonth() + 1).padStart(2, '0');
+			const day = String(date.getDate()).padStart(2, '0');
+			return `${year}-${month}-${day}`;
+		}
+
+		// TODO
+		const stream = await createStream(
+			[
+				{ role: 'system', content: 'Your task is to create a short and simple file name given its content.' },
+				{ role: 'system', content: 'You need to answer only the file name. No other text or decoration needed.' },
+				{ role: 'system', content: 'File name must not contain whitespaces and extension.' },
+				{ role: 'user', content: `Content:\n${textEditor.document.getText()}` }
+			] as OpenAIChatMessage[], {} as OpenAI.ChatCompletionCreateParamsStreaming);
+		let filename = "";
+		for await (const chunk of stream) {
+			const chunkText = chunk.choices[0]?.delta?.content || '';
+			filename += chunkText;
+		}
+
+		const configuration = vscode.workspace.getConfiguration();
+		let filepath = configuration.get<string>("markdown.copilot.instructions.autonameFilepath");
+		if (filepath === undefined || filepath.trim().length === 0) {
+			filepath = "memo/${date}/${filename}.md";
+		}
+		filepath = filepath.replace("${date}", getDate());
+		filepath = filepath.replace("${filename}", filename);
+
+		const destUri = vscode.Uri.joinPath(vscode.workspace.workspaceFolders![0].uri, filepath);
+		vscode.workspace.fs.writeFile(destUri, Buffer.from(textEditor.document.getText()))
```

Comment:
Now users can edit suggested filename. Are you sure users want to edit the whole file *path*?

![](https://github.com/kurusugawa-computer/markdown-copilot-vscode/assets/9913176/87dee122-fb4b-4a5d-8ac2-f9e61b4c10bc)

Reply:
Is it possible to display a [`file selection dialog`](https://www.google.com/search?q=file+selection+dialog&udm=2) instead of `vscode.window.showInputBox`?

If there are no intermediate directories, they should be created temporarily, and the directories should be purged when (the save is cancelled and there are no files in the intermediate directories).

Reply:
Maybe we can use `workbench.action.files.save` or `workbench.action.files.saveAs`.
```javascript
let uri = Uri.file('/some/path/to/folder');
let success = await commands.executeCommand('workbench.action.files.saveAs', uri);
```

Reply:
It seems the second argument `uri` is ignored.

Reply:
`showSaveDialog` is promising.